wiki/Gameplay/Judgement/osu!/en.md

```diff
@@ -1,41 +1,62 @@
+- If [ScoreV2](/wiki/Gameplay/Game_modifier/ScoreV2) is active, the accuracy of hitting the slider head is additionally accounted for when judging the entire slider:
+  - Getting a 300 for a slider requires getting a 300 judgement for the slider head.
+  - Getting a 100 for a slider requires getting a 100 judgement or better for the slider head.
+
+The slider parts also have additional influence on the current [combo](/wiki/Gameplay/Combo_(score_multiplier)):
+
+  - Tapping the slider head too early does not incur a miss, but will reset combo to 0.
+  - Missing a slider tick or repeat does not incur a miss, but will reset combo to 0.
+  - Missing the slider end does not incur a miss, but will not increment combo.
+
+### Spinners
+
+Each spinner has a set number of spins required to complete it.
+This number depends on the [overall difficulty](/wiki/Beatmap/Overall_difficulty#sliders-and-spinners) of the beatmap.
 
-- A 300, 100 or 50 given from a normal hit circle depending on how accurately the player tap on it.
-- A miss given if the player do not click on the hit circle at all or click on it way too early.
+- A 300 is given if the player completes the number of spins required for completion, or more.
+- A 100 is given if the player completes one spin less than the required number.
+- A 50 is given if the player completes at least 25% of the required spins.
+- If the player fails to meet the minimum of 25% of spins required, the spinner is judged as a miss.
 
-**Sliders:**
+For some very short spinners, the number of spins required is actually calculated to be 0, and thus the spinner always automatically completes itself with a 300.
 
-- The judgement for sliders does not depend on the accuracy of the initial hit.
-- Sliders consist of slider ticks, which include the start, end and repeat points of the slider.
-- A 300 given if the player successfully gets all slider ticks, a 100 given if the player gets at least half of them, and a 50 given if the player manages to collect even one of the ticks.
-- Tapping a slider **too early** does not incur a miss but will **reset** the score multiplier to **0**.
-  - Not collecting the slider ticks, shown or otherwise, will not incur a miss, but will **reset** the score multiplier to **0**.
-  - Missing the slider end does not incur a miss but will **not** increment the score multiplier.
-- A miss given if the player collects none of the slider ticks.
+#### History
 
-**Spinner:**
+The spinner judging algorithm has been significantly changed in the [20190510.2 Cutting Edge release](https://osu.ppy.sh/home/changelog/cuttingedge/20190510.2).
+Prior to the change, non-miss judgements were significantly harder to achieve, as the difference between a 50 and a 100, as well as between a 100 and a 300, was equal to a half of a spin.
+Some beatmaps would even feature spinners that could never be judged with a 300, as they were too short to complete fully.
```

Comment:
this should be written all on the same line, like in most, if not all articles. besides, the linked changelog entry may be a good candidate for a footnote

Reply:
Yeah I don't know what the policy on line breaks is. I'm mostly putting one sentence per line but I'll stop doing so if it's not proper.

I can make this a footnote, but what do I attach it to? The section header?

Reply:
I think the release name fits the most:

```md
The spinner judging algorithm has been significantly changed in the 20190510.2 Cutting Edge release.[^ce-spinners-ref]

## References

[^ce-spinners-ref]: [Cutting Edge 20190510.2 · changelog](https://osu.ppy.sh/home/changelog/cuttingedge/20190510.2)
```

Reply:
Oh you mean footnoting just the link to the changelog rather than the entire section? Sure I can do that.

Reply:
why do a footnote if you're duplicating the "20190510.2 Cutting Edge release"?

Reply:
I... yeah, this is excessive, disregard the footnote part of my suggestion. seeing everything as a nail now that we've got a hammer.

Reply:
Line breaks have been culled in [c19be0f](https://github.com/ppy/osu-wiki/pull/7527/commits/c19be0fdb61975311e14333ff5b427b6485542c4).